.github/workflows/mitm.yml

```diff
@@ -9,6 +9,10 @@ on:
       - synchronize
   workflow_dispatch:
 
+concurrency:
+  group: ${{ github.event_name }}-${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }} # this will cancel only PRs
```

Comment:
We can simplify here by removing the `workflow_dispatch` trigger (it's currently not used) and changing `cancel-in-progress` to `true`.